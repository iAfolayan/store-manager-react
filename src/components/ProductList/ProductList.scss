$primarycolor: #0f488a;

.card {
  margin: 20px 0px;
}

.product-title{
	padding-top: 15px;
	text-align:center;
}

.img-holder{
	max-height: 250px;
	max-width: 200px;
	margin: 0 auto;
	margin-top: 7px;
	height:250px;
	position: relative;
	text-align: center;
}

.img-holder img{
	max-height: 201px;
	max-width: 200px;
	margin: 0 auto;
	margin-top: 7px;
	min-height: 25px;
	min-width: auto;
}

.img-overlay{
	max-height: 91.2px;
	max-width: 200px;
	position: absolute;
	right:0;
	left: 0;
	bottom: -7px;
	overflow: hidden;
	height: 0%; 
	transition: height 0.5s;
	background: $primarycolor;
	color: white;
	display: flex;
	align-items:center;
	justify-content:center;
}

.card:hover .img-overlay{
	height: 100%;
}
.card:hover{
	box-shadow: 0 2px 21px 0 rgba(0,0,0,0.19);
}

.card .dets{
	max-height: 70px;
	max-width:200px;
	margin:15px auto;
	display: flex;
	align-items: center;
	justify-content: space-between;
	margin-bottom: 25px;
	text-align: center;
}
.dets {
		width: 100%;
    margin: 0;
}

.card a {
	text-decoration: none;
	color: #000;
}

.product-holder-prot .dets {
	margin: auto;
	text-align: center;	
}
article {
	text-align: center;
    font-size: 14px;
    line-height: 25px;
    padding-right: 5px;
    padding-left: 5px;
}

.img-holder h3 {
	text-align: center;
	font-size: 13px;
}