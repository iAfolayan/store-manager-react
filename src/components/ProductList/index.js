import React from 'react';
import { Link } from 'react-router-dom';
import { Card, CardImg } from 'reactstrap';
import './ProductList.scss';

const ProductList = ({ products }) => {
	const productList = products.length ? (
		products.map((product) => {
			return (
				<div className="col-md-3 float-left" key={product.id}>
					<Card className="card">
						<h5 className="card-header text-center">
							<strong>{product.name}</strong>
						</h5>
						<Link to="">
							<div className="img-holder">
								<CardImg className="cart-image" src={product.image} alt="" />
								<div className="img-overlay">
									<article>{product.description.substring(0, 40)}</article>
								</div>
								<h3>&#8358;{product.price} / piece</h3>
							</div>
						</Link>
						<div className="dets">
							{' '}
							<small>Minimum Qty {product.minimumallowed}</small>
						</div>
					</Card>
				</div>
			);
		})
	) : (
		<p className="mx-auto">You don't have any product to display.</p>
	);
	return <div>{productList}</div>;
};

export default ProductList;
