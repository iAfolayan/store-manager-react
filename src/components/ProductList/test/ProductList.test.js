import React from 'react';
import { shallow } from 'enzyme'
import ProductList from '../index';

describe('<ProductList />', () => {
  test('should test product info', () => {
    const wrapper = shallow(<ProductList />);
    expect(wrapper).toMatchSnapshot();
  });
});