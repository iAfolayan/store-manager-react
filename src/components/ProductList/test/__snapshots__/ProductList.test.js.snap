// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> should test product info 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
